src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlObjectSerializerReadContextComplex.cs

```diff
@@ -136,7 +136,9 @@ private object InternalDeserializeWithSurrogate(XmlReaderDelegator xmlReader, Ty
 
         private Type ResolveDataContractTypeInSharedTypeMode(string assemblyName, string typeName, out Assembly assembly)
         {
-            throw new PlatformNotSupportedException();
+            // The method is used only when _mode == SerializationMode.SharedType.
+            // _mode is set to SerializationMode.SharedType only when the context is for NetDataContractSerializer.
+            throw new PlatformNotSupportedException(); // Not supported because NetDataContractSerializer is not supported on NetCore.
```

Comment:
Could you please add messages to this PNSE also, and the other three below? All of them seem like "interesting reasons" that it would be helpful for the caller to get as a message.